src/webpimage.cpp

```diff
@@ -508,10 +518,12 @@ namespace Exiv2 {
 #endif
 
         chunkId.pData_[4] = '\0' ;
-        while ( !io_->eof() && (uint64_t) io_->tell() < filesize) {
-            io_->read(chunkId.pData_, WEBP_TAG_SIZE);
-            io_->read(size_buff, WEBP_TAG_SIZE);
-            const uint32_t size = Exiv2::getULong(size_buff, littleEndian);
+        while (!io_->eof() && io_->tell() < filesize) {
+            readOrThrow(*io_, chunkId.pData_, WEBP_TAG_SIZE, Exiv2::kerCorruptedMetadata);
+            readOrThrow(*io_, size_buff, WEBP_TAG_SIZE, Exiv2::kerCorruptedMetadata);
+            const long size = Exiv2::getULong(size_buff, littleEndian);
+            enforce(0 <= size, Exiv2::kerCorruptedMetadata);
```

Comment:
See my comment in #813, this is relying on undefined behavior. To properly check this, you must do something like this:
``` C++
const uint32_t size = Exiv2::getULong(size_buff, littleEndian);
enforce(size < std::numeric_limits<long>::max(), Exiv2::kerCorruptedMetadata);
const long long_size = size;
```

Reply:
done